### PR TITLE
[RA-5096] Implement support of the Linux Kernel v6.8

### PR DESCRIPTION
**Pull request by stanislavbarantsev**

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -221,6 +221,46 @@ static struct block_device *blkdev_get_by_path(const char *path, fmode_t mode, v
...
 
+static int elastio_snap_freeze_bdev(struct block_device *bdev)
+{
+	int ret;
+
+#ifdef HAVE_THAW_BDEV_INT
+	sb = freeze_bdev(bdev);
+	ret = (IS_ERR(sb)) ? PTR_ERR(sb) : 0;
+#else
+#ifdef HAVE_BDEV_FREEZE
```
Comment (kgermanov) [inline src/main.c]: Lets use 
`#elif HAVE_BDEV_FREEZE`
For reduce nested

File: src/main.c [hunk 2]
```diff
@@ -221,6 +221,46 @@ static struct block_device *blkdev_get_by_path(const char *path, fmode_t mode, v
...
+	return ret;
+}
+
+static int elastio_snap_thaw_bdev(struct block_device *bdev)
+{
+	int ret;
+#ifdef HAVE_THAW_BDEV_INT
+	ret = thaw_bdev(bdev, sb);
+#else
+#ifdef HAVE_BDEV_FREEZE
```
Comment (kgermanov) [inline src/main.c]: Lets use 
`#elif HAVE_BDEV_FREEZE`
For reduce nested

File: src/main.c [hunk 3]
```diff
@@ -528,6 +569,9 @@ static int elastio_snap_should_remove_suid(struct dentry *dentry)
 	#define elastio_snap_bio_bi_disk(bio) ((bio)->bi_disk)
 #endif
 
+#if defined HAVE_BDEV_OPEN_BY_PATH
+	#define elastio_snap_bdev_release(bd_handle) bdev_release(bd_handle);
+#else
```
Comment (kgermanov) [inline src/main.c]: Lets use 
`#elif HAVE_BLKDEV_PUT_1`
For reduce nested

File: src/main.c [hunk 4]
```diff
@@ -4752,17 +4791,34 @@ static void __tracer_destroy_base_dev(struct snap_device *dev){
...
+	dev->sd_bd_handle = elastio_snap_bdev_open_by_path(bdev_path, FMODE_READ, NULL);
+	if (!IS_ERR(dev->sd_bd_handle)) {
+		dev->sd_base_dev = dev->sd_bd_handle->bdev;
+	} else {
+		ret = PTR_ERR(dev->sd_bd_handle);
+		dev->sd_base_dev = NULL;
+		LOG_ERROR(ret, "error finding block device '%s'", bdev_path);
+		goto error;
+	}
+#endif
```
Comment (kgermanov) [inline src/main.c]: There are a lot of ifdev around this function. May be do some this:
```
struct res_blkdev {
   *sd_base_dev;
#ifdef HAVE_BDEV_OPEN_BY_PATH
   *sd_bd_handle;
#endif
};

// Incapsulate all defs in this function
res_blkdev* elastio_snap_blkdev_get_by_path(const char *path, fmode_t mode, void *holder)
{
#if defined HAVE_BDEV_OPEN_BY_PATH
    res_blkdev res{NULL, NULL};
	res.sd_bd_handle = bdev_open_by_path(path, mode, holder, NULL);
	if (IS_ERR(dev->sd_bd_handle)) {
		return NULL;
	}
	res.sd_base_dev = res.sd_bd_handle->bdev;
	return res;
#elif HAVE_BLKDEV_GET_BY_PATH_4
	return res_blkdev{blkdev_get_by_path(path, mode, holder, NULL), NULL};
#else
	return res_blkdev{blkdev_get_by_path(path, mode, holder), NULL};
#endif
  
}

// Incapsulate all releases
void elastio_snap_bdev_release(struct res_blkdev* blkdev)
{
#ifdef HAVE_BDEV_OPEN_BY_PATH
	bdev_release(blkdev->sd_bd_handle);
#elif HAVE_BLKDEV_PUT_1
//#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,28)
	blkdev_put(blkdev->sd_base_dev);
#elif defined HAVE_BLKDEV_PUT_HOLDER
	blkdev_put(blkdev->sd_base_dev, NULL);
#else
	blkdev_put(blkdev->sd_base_dev, FMODE_READ);
#endif
}
....


// Usage:
dev->res_base_dev = elastio_snap_blkdev_get_by_path(bdev_path, FMODE_READ, NULL);

void some_function(dev->res_base_dev->sd_base_dev);
..
elastio_snap_bdev_release(dev->res_base_dev);
```

**Review by kgermanov — CHANGES_REQUESTED** [review]

File: src/main.c (hunk 4 above)
Comment (stanislavbarantsev) [reply src/main.c]: Refactored in a similar way

**Review by kgermanov — APPROVED** [review]

**Review by AnatolyZ — APPROVED** [review]

**Pull request merged**